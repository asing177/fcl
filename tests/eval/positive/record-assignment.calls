updateName("robert")
updateStreet("posh st")
