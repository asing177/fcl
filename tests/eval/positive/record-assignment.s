type Address = Addr(text street, text city);
type Person = Pers(text name, Address address);

Person bob = Pers("bob", Addr("main st", "springfield"));

@initial
updateName(text newName) {
  bob.name = newName;
  stay()
}

@initial
updateStreet(text newStreet) {
  bob.address.street = newStreet;
  stay()
}

@initial
finish() {
  terminate()
}
