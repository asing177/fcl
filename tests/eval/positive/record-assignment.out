*** Deltas ***
[ ModifyGlobal
    Name { unName = "bob" }
    VConstr
      { vConstrName = MkNameUpper "Pers"
      , vConstrParams =
          [ VText "robert"
          , VConstr
              { vConstrName = MkNameUpper "Addr"
              , vConstrParams = [ VText "main st" , VText "springfield" ]
              }
          ]
      }
, ModifyGlobal
    Name { unName = "bob" }
    VConstr
      { vConstrName = MkNameUpper "Pers"
      , vConstrParams =
          [ VText "robert"
          , VConstr
              { vConstrName = MkNameUpper "Addr"
              , vConstrParams = [ VText "posh st" , VText "springfield" ]
              }
          ]
      }
]

*** Globals ***
fromList
  [ ( Key { unKey = "bob" }
    , VConstr
        { vConstrName = MkNameUpper "Pers"
        , vConstrParams =
            [ VText "robert"
            , VConstr
                { vConstrName = MkNameUpper "Addr"
                , vConstrParams = [ VText "posh st" , VText "springfield" ]
                }
            ]
        }
    )
  , ( Key { unKey = "testAddr" }
    , VAccount (Address "fwBVDsVh8SYQy98CzYpNPcbyTRczVUZ96HszhNRB8Ve")
    )
  , ( Key { unKey = "testAddr2" }
    , VAccount (Address "Fw68hsis6JPW4ed3ER1VC959TTf9DqberLxrq5uECKxh")
    )
  , ( Key { unKey = "testAddr3" }
    , VAccount (Address "4Stv5gWGt78D3p919HFcKEschSmyGKGrAX2G35xJgdhw")
    )
  ]
